src/server/api/controllers/user.controller.js

```diff
@@ -0,0 +1,113 @@
+const knex = require('../../config/db');
+const Error = require('../lib/utils/http-error');
+
+const getUserById = async (id, role) => {
+  try {
+    const eventUsers = await knex.raw(
+      'SELECT ' +
+        'ev.id AS eventId, ' +
+        'org.name AS organization, ' +
+        'ev.event_date AS eventDate, ' +
+        'ev.created_at AS eventTime, ' +
+        'WEEK(ev.created_at) as eventWeek ' +
+        'FROM ' +
+        'events_users AS eu, ' +
+        'users AS us, ' +
+        'events_students AS es, ' +
+        'organizations AS org, ' +
+        'events AS ev ' +
+        'WHERE ' +
+        'us.id = eu.users_id ' +
+        'AND es.student_id = eu.id ' +
+        'AND org.id = us.organization_id ' +
+        'AND ev.id = es.event_id ' +
+        'AND eu.users_id = ? ' +
+        'limit 1;',
+      [id],
+    );
+    if (eventUsers.length === 0) {
+      return new Error(`No Entries with the id of ${id}`, 404);
+    }
+    const usableData = eventUsers[0][0];
```

Comment:
What is `usableData` here? Is it an object, a string, or a number?

Reply:
knex raw returns the data in a format of array of arrays. The first array is the result but the second one is the column metadata, which we dont need. Thats why I have written it like this, I will try to refine it.